### ISSUE TITLE
Default calendar has been changed / Let the user choose the permanent default calendar

### ISSUE THREAD

**Issue by AND51** [nextcloud/calendar#360]
Hello,

this is my first bug report here. If I can improve something, I appreciate any feedback.

### Problem

I have 3 calendars for a long time (since NC 9 and earlier). The calendar "Persönlich" (private) was always the default calendar. I add most events here. My other calendars are "ma*****" (work) and "Gemeinsamer Kalender" (Shared calendar).
Somehow, the "shared calendar" became the new default, so that I have to change the calendar when adding a new event. I don't know how the exact cause, I assume it's caused by one of the last Nextcloud-upgrades.

![screenshot-2017-03-09-10 13 53](https://cloud.githubusercontent.com/assets/26226303/23744114/561838d6-04b3-11e7-86bb-07e4943ed528.png)

### Steps to reproduce
1. Login to Nextcloud.
2. Click "Calendar".
3. Have multiple calendars, not just one.
4. Create a new event, no matter if simple or advanced event editor pops up.
5. Click drop down to choose the calendar the new event is being assinged to.

### Expected behaviour
My calendar "Persönlich" (Personal) should be the default calendar and be the first item. Need to select this calendar every time when adding events to personal calendar. Nowadays, another calendar became the first item.
![screenshot-2017-03-09-11 18 54](https://cloud.githubusercontent.com/assets/26226303/23745895/42b4296a-04ba-11e7-870f-a467662612ff.png)

### Actual behaviour
I used to see personal calendar as default, i. e. the first item is "Personal". No need to extra select this calendar where I add 90% of my events.

### Server configuration
**Operating system**: Ubuntu 16.04.1 LTS

**Web server:** Apache 2.4.18 (Ubuntu)

**Database:** mysql  Ver 15.1 Distrib 10.0.28-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

**PHP version:** PHP-FPM 7.0.13

**Server version:** 11.0.2

**Calendar version:** 1.5.1 (I updated the calender to this version, the problem exists longer)

**Updated from an older installed version or fresh install:** Yes, since OC 8. Everything worked fine up to NC 10 and (I think) up to NC 11.0

**Signing status (ownCloud/Nextcloud 9.0 and above):** Sorry, what do you mean by this?

```
Login as admin user into your cloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

No errors have been found.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your instance's installation folder

Enabled:
  - activity: 2.4.1
  - admin_audit: 1.1.0
  - calendar: 1.5.1
  - comments: 1.1.0
  - contacts: 1.5.3
  - dav: 1.1.1
  - direct_menu: 0.10.0 (problem exists before activating this app)
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_automatedtagging: 1.1.1
  - files_external: 1.1.2
  - files_pdfviewer: 1.0.1
  - files_retention: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - firstrunwizard: 2.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - previewgenerator: 1.0.2
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - systemtags: 1.1.3
  - templateeditor: 0.2
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - encryption
  - external
  - files_accesscontrol
  - user_external
  - user_ldap
  - user_saml
```

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your instance's installation folder

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/data",
        "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***",
        "version": "11.0.2.7",
        "dbtype": "mysql",
        "dbname": "nextcloud",
        "dbhost": "10.0.8.37",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Berlin",
        "installed": true,
        "cron_log": true,
        "log_rotate_size": 20971520,
        "default_language": "de",
        "defaultapp": "files,calendar,contacts,gallery",
        "enable_avatars": true,
        "allow_user_to_change_display_name": true,
        "skeletondirectory": "",
        "remember_login_cookie_lifetime": 1296000,
        "session_lifetime": 259200,
        "session_keepalive": true,
        "auth.bruteforce.protection.enabled": true,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "nextcloud",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtptimeout": 30,
        "mail_smtpsecure": "tls",
        "htaccess.RewriteBase": "\/",
        "trashbin_retention_obligation": "auto",
        "versions_retention_obligation": "auto",
        "enable_previews": true,
        "preview_max_x": 2048,
        "preview_max_y": 2048,
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "preview_office_cl_parameters": " --headless --nologo --nofirststartwizard --invisible --norestore --convert-to pdf --outdir ",
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\Illustrator",
            "OC\\Preview\\Movie",
            "OC\\Preview\\MSOffice2003",
            "OC\\Preview\\MSOffice2007",
            "OC\\Preview\\MSOfficeDoc",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\PDF",
            "OC\\Preview\\Photoshop",
            "OC\\Preview\\StarOffice",
            "OC\\Preview\\SVG",
            "OC\\Preview\\TIFF",
            "OC\\Preview\\Font",
            "OC\\Preview\\MarkDown"
        ],
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\APCu",
        "redis": {
            "host": "\/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/tmp\/redis.sock",
            "port": 0,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***",
            "dbindex": 0
        },
        "tempdirectory": "\/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/tmp",
        "hashingCost": 10,
        "blacklisted_files": [
            ".htaccess"
        ],
        "asset-pipeline.enabled": true,
        "filesystem_check_changes": 1,
        "maintenance": false,
        "loglevel": 1,
        "has_internet_connection": true,
        "check_for_working_webdav": true,
        "check_for_working_wellknown_setup": true,
        "check_for_working_htaccess": true,
        "config_is_read_only": false,
        "appcodechecker": true,
        "updatechecker": true,
        "updater.release.channel": "stable",
        "appstoreenabled": true,
        "theme": ""
    }
}

```

**Are you using external storage, if yes which one:** One external WebDAV

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 49, Vivaldi 1.7.735.46 (I assume it's not browser related)

**Operating system:** Errrr... Linux Mint based on 16.04?

**CalDAV-clients:** iPhone with iOS 10.3

#### Web server error log
```
[Wed Mar 08 07:40:13.776699 2017] [proxy_fcgi:error] [pid 29896:tid 139657270310656] (104)Connection reset by peer: [client 10.0.8.29:2670] AH01075: Error dispatching request to : 
[Thu Mar 09 00:07:31.030980 2017] [proxy_fcgi:error] [pid 29999:tid 139657614411520] [client 10.0.8.29:38712] AH01067: Failed to read FastCGI header
[Thu Mar 09 00:07:31.030994 2017] [proxy_fcgi:error] [pid 29999:tid 139657614411520] (104)Connection reset by peer: [client 10.0.8.29:38712] AH01075: Error dispatching request to : 
[Thu Mar 09 03:28:21.248375 2017] [proxy_fcgi:error] [pid 29896:tid 139657656375040] [client 10.0.8.29:42218] AH01067: Failed to read FastCGI header
[Thu Mar 09 03:28:21.248389 2017] [proxy_fcgi:error] [pid 29896:tid 139657656375040] (104)Connection reset by peer: [client 10.0.8.29:42218] AH01075: Error dispatching request to :
```
I assume this is not relevant.

### tail -n 10
```
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Undefined index: templateeditor at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Undefined index: theming at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Undefined index: twofactor_backupcodes at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Undefined index: updatenotification at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Undefined index: workflowengine at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
```
### grep -i calendar
```
{"reqId":"WMA-Z38AAQEAAHTIq4YAAAAj","remoteAddr":"10.0.1.235","app":"admin_audit","message":"Login successful: \"Melina\"","level":1,"time":"2017-03-08T18:29:12+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Melina\/inbox\/","user":"Melina","version":"11.0.2.7"}
{"reqId":"WMByNH8AAQEAAHTIrOoAAAAL","remoteAddr":"10.0.1.235","app":"admin_audit","message":"Login attempt: \"Melina\"","level":1,"time":"2017-03-08T22:05:56+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Melina\/","user":"--","version":"11.0.2.7"}
{"reqId":"WMByNH8AAQEAAHTIrOoAAAAL","remoteAddr":"10.0.1.235","app":"admin_audit","message":"Login successful: \"Melina\"","level":1,"time":"2017-03-08T22:05:57+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Melina\/","user":"Melina","version":"11.0.2.7"}
{"reqId":"WMByNn8AAQEAAHTIrOsAAAAI","remoteAddr":"10.0.1.235","app":"admin_audit","message":"Login attempt: \"Melina\"","level":1,"time":"2017-03-08T22:05:58+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Melina\/inbox\/","user":"--","version":"11.0.2.7"}
{"reqId":"WMByNn8AAQEAAHTIrOsAAAAI","remoteAddr":"10.0.1.235","app":"admin_audit","message":"Login successful: \"Melina\"","level":1,"time":"2017-03-08T22:05:59+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Melina\/inbox\/","user":"Melina","version":"11.0.2.7"}
{"reqId":"WMET@38AAQEAABEJdnEAAAAT","remoteAddr":"***REMOVED SENSITIVE VALUE***","app":"admin_audit","message":"Login attempt: \"Andre\"","level":1,"time":"2017-03-09T09:36:11+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Andre\/","user":"--","version":"11.0.2.7"}
{"reqId":"WMET@38AAQEAABEJdnEAAAAT","remoteAddr":"***REMOVED SENSITIVE VALUE***","app":"admin_audit","message":"Login successful: \"Andre\"","level":1,"time":"2017-03-09T09:36:12+01:00","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Andre\/","user":"Andre","version":"11.0.2.7"}
{"reqId":"WMET-H8AAQEAABEJdnIAAAAU","remoteAddr":"***REMOVED SENSITIVE VALUE***","app":"admin_audit","message":"Login attempt: \"Andre\"","level":1,"time":"2017-03-09T09:36:12+01:00","method":"PUT","url":"\/remote.php\/dav\/calendars\/Andre\/persnlich\/6E7411C8-8D6E-404B-8B0F-1A439FE29955.ics","user":"--","version":"11.0.2.7"}
{"reqId":"WMET-H8AAQEAABEJdnIAAAAU","remoteAddr":"***REMOVED SENSITIVE VALUE***","app":"admin_audit","message":"Login successful: \"Andre\"","level":1,"time":"2017-03-09T09:36:12+01:00","method":"PUT","url":"\/remote.php\/dav\/calendars\/Andre\/persnlich\/6E7411C8-8D6E-404B-8B0F-1A439FE29955.ics","user":"Andre","version":"11.0.2.7"}
{"reqId":"G8wwdnpvukeUxajWFbzy","remoteAddr":"","app":"PHP","message":"Undefined index: calendar at \/srv\/www\/vhosts\/***REMOVED SENSITIVE VALUE***\/htdocs\/lib\/private\/AppConfig.php#292","level":3,"time":"2017-03-09T10:56:50+01:00","method":"--","url":"--","user":"--","version":"11.0.2.7"}
```

#### Browser log
```
Everything is HTTP Code 200 when accessing calendar, nothing mysterious here.
```

#### SELECT * FROM oc_calendars \G
```
*************************** 1. row ***************************
           id: 3
 principaluri: principals/system/system
  displayname: Contact birthdays
          uri: contact_birthdays
    synctoken: 1366
  description: NULL
calendarorder: 0
calendarcolor: #FFFFCA
     timezone: NULL
   components: VEVENT
  transparent: 0
*************************** 2. row ***************************
           id: 4
 principaluri: principals/users/Hans
  displayname: default
          uri: default
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: NULL
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 3. row ***************************
           id: 6
 principaluri: principals/users/Andre
  displayname: Persönlich
          uri: persnlich
    synctoken: 807
  description: NULL
calendarorder: 4
calendarcolor: #1BADF8
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 4. row ***************************
           id: 8
 principaluri: principals/users/Andre
  displayname: Gemeinsamer Kalender
          uri: gemeinsamer-kalender
    synctoken: 151
  description: NULL
calendarorder: 2
calendarcolor: #ffa500
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 5. row ***************************
           id: 9
 principaluri: principals/users/Andre
  displayname: ma*********
          uri: ma*********
    synctoken: 165
  description: NULL
calendarorder: 3
calendarcolor: #b3dc6c
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 6. row ***************************
           id: 10
 principaluri: principals/users/Andre
  displayname: Contact birthdays
          uri: contact_birthdays
    synctoken: 71
  description: NULL
calendarorder: 0
calendarcolor: #FFFFCA
     timezone: NULL
   components: VEVENT
  transparent: 0
*************************** 7. row ***************************
           id: 11
 principaluri: principals/users/system
  displayname: Personal
          uri: personal
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: NULL
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 8. row ***************************
           id: 12
 principaluri: principals/users/craemer
  displayname: Personal
          uri: personal
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: NULL
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 9. row ***************************
           id: 14
 principaluri: principals/users/Melina
  displayname: Personal
          uri: personal
    synctoken: 47
  description: NULL
calendarorder: 0
calendarcolor: #FF2968
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 10. row ***************************
           id: 15
 principaluri: principals/users/Michael
  displayname: Personal
          uri: personal
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: NULL
     timezone: NULL
   components: VEVENT,VTODO
  transparent: 0
*************************** 11. row ***************************
           id: 16
 principaluri: principals/users/Melina
  displayname: Contact birthdays
          uri: contact_birthdays
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: #FFFFCA
     timezone: NULL
   components: VEVENT
  transparent: 0
*************************** 12. row ***************************
           id: 17
 principaluri: principals/users/Michael
  displayname: Contact birthdays
          uri: contact_birthdays
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: #FFFFCA
     timezone: NULL
   components: VEVENT
  transparent: 0
*************************** 13. row ***************************
           id: 18
 principaluri: principals/users/craemer
  displayname: Contact birthdays
          uri: contact_birthdays
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: #FFFFCA
     timezone: NULL
   components: VEVENT
  transparent: 0
*************************** 14. row ***************************
           id: 19
 principaluri: principals/users/system
  displayname: Contact birthdays
          uri: contact_birthdays
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: #FFFFCA
     timezone: NULL
   components: VEVENT
  transparent: 0
14 rows in set (0.01 sec)
```

Phew, that was pretty much. Let me know if you need futher information. I appreciate any immediate/manual fix. But if you consider this a bug, I can wait for the next update and then report if that update helped. Moreover, it would be nice to have the possibility to choose the default calendar somewhere in the app's settings (feature request). Consider this a side note. Of course, I can open an extra issue for this, if you wish.

**Comment by georgehrke** [comment]
related to https://github.com/nextcloud/calendar/issues/9

This will allow you to chose any calendar as your default one.

**Comment by georgehrke** [comment]
> Nowadays, another calendar became the first item.

When did this change? 
Just from one night to another or after an update?

**Comment by AND51** [comment]
Didn't notice #9, because I searched for "default calendar" instead of "sortable calendar". But yes, my wish is related to that ticket.

> When did this change?
I've thought about this intensively. Unfortunately I cannot remember the exact cause, because the calendar switch happened a few weeks ago. From my experience I conclude that one of the following options must be responsible:
- An update within the NC 11.x.x-branch (I only use stable/regular versions)
- The fact, that I removed a 4th calendar "Fristen" (deadlines) long time ago.

Now that I read the SELECT in my inital post again, I notice the columns 'calendarid' and 'calendarorder'. I know that all calendars I ever had were created in the same order as 'calendarid':
1. Persönlich (calendarid 6, calendarorder 4)
2. Gemeinsamer Kalender (calendarid 8, calendarorder 3)
3. ma****** (calendarid 9, calendarorder 2)
4. Fristen (calendarid unknown, calendarorder unknown)

Obviously, the calendarid-order does not match the the calendarorder-order (anymore?). But I have no idea why.

**Comment by AND51** [comment]
Question: May I manipulate the column 'calendarorder' by hand or do you not recommend this? Also if you prefer to examine this further without my intervention thats fine. I can endure this "bug".

**Comment by georgehrke** [comment]
You said you setup the calendar on iOS. I suspect that iOS manipulated these calendarorder values.
iOS can't set the calendarorder property for the shared calendar (yet), that's why that order is left at 0.

> May I manipulate the column 'calendarorder' by hand or do you not recommend this? 

Yes, you can just manipulate them. Either you set all back to 0 (not sure if iOS is gonna try to set them again) or set the shared calendar to something higher than the other ones.

**Comment by georgehrke** [comment]
I'm closing this issue because:
- creating your own default calendar will be fixed with #9 
- it is the expected behavior, that changes in the iOS calendar order will be visible in the calendar app
- the bug, that the calendar order can't be set for shared calendars, is related to https://github.com/nextcloud/server/issues/1463 and will be fixed with Nextcloud 12.

**Comment by AND51** [comment]
> I suspect that iOS manipulated these calendarorder values.

Seems like iOS 10.3 on iPhone 7 does **not** manipulate orders. I've verified this by executing these steps:

1. Set **each** 'calendarorder' to zero (0, not NULL of course).
2. Hit Ctrl+F5 in calendar app. (Effect: the order of the calendars is now "Persönlich", "Gemeinsamer Kalender" and "ma****" in NC calendar app. "Persönlich" is now the default when creating an event. Yay!)
3. Deactivated the CalDAV-account in iOS.
4. Changed color of calendar "Persönlich" in NC.
5. Activated the CalDAV-account in iOS.
6. Played around with the subscriptions in iOS and pulled down in settings-dialogue to force instant sync.
7. Check iPhone Settings→Calendar→Default Calendar is set to "ma****".
8. Changed iPhones default calendar to "Persönlich".
9. Forced sync/reload just like in step 6.
10. Created calendar "Georg" via iPhone. (Effect: A new 15th row in MySQL, look below.)
11. Deleted "Georg" via Nextcloud.
12. Added and deleted "Georg2" via Nextcloud.


```
*************************** 15. row ***************************
           id: 20
 principaluri: principals/users/Andre
  displayname: Georg
          uri: 14E321D8-5072-4244-8D7E-86BE8D06B04D
    synctoken: 1
  description: NULL
calendarorder: 0
calendarcolor: #FF2968
     timezone: BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
BEGIN:VTIMEZONE
TZID:Europe/Berlin
BEGIN:DAYLIGHT
TZOFFSETFROM:+0100
RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU
DTSTART:19810329T020000
TZNAME:MESZ
TZOFFSETTO:+0200
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:+0200
RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU
DTSTART:19961027T030000
TZNAME:MEZ
TZOFFSETTO:+0100
END:STANDARD
END:VTIMEZONE
END:VCALENDAR

   components: VEVENT
  transparent: 1

```
Well, on calendar creation via iPhone the column 'timezone' contains much more text (and even multi-line text) compared to calendars created cirectly in NC.


**All in all,** from what I tested the iPhone cannot/does not manipulate the order. (Didn't check via tcpdump/proxy though.) The column 'calendarorder' is zero everywhere. I can live with the current situation as "Persönlich" is now my default again. Let me know, if you want me to test further things.

**Moreover,** you cannot change the order in the iPhone or from within the iPhone. You can only check/uncheck a calendar. The only thing you can do is to set a default calendar, but the iPhone's calendar has nothing to do with NC's default calendar. I think, the iPhone saves its default locally.

**Apart from that** I agree with you. I'm looking forward to #9 and nextcloud/server#1463. I hope I could enrich you with my iOS test results. Thank you for your attention.

**Comment by georgehrke** [comment]
> Moreover, you cannot change the order in the iPhone or from within the iPhone. You can only check/uncheck a calendar.

Oh. I think they changed that(, but you can still change the order on macOS calendar.app.) I suspect that iOS sets the calendar order when setting up the CalDAV account initially.

>  The only thing you can do is to set a default calendar, but the iPhone's calendar has nothing to do with NC's default calendar. I think, the iPhone saves its default locally.

Yes, I think so too. In the Nextcloud calendar, the first one just happens to be your default.

**Comment by AND51** [comment]
> I suspect that iOS sets the calendar order when setting up the CalDAV account initially.

Thehe… I avoided this test, because I didn't want to re-enter my long password. Nevertheless I revised that for you :-)
Result: Neither the deletion did change the 'calendarorder' nor did the re-creation of the CalDAV-account. Also other actions like changing the 'calendarcolor' from within the iPhone did not change the 'calendarorder'.

Dunno if it matters, but on iPhone I only use the long URL https://BEEP/remote.php/dav/principals/users/Andre/ and not the other one.
